src/Symfony/Component/EventDispatcher/ContainerAwareEventDispatcher.php

```diff
@@ -105,7 +105,7 @@ public function removeListener($eventName, $listener)
     public function hasListeners($eventName = null)
     {
         if (null === $eventName) {
-            return (bool) count($this->listenerIds) || (bool) count($this->listeners);
+            return $this->listenerIds || $this->listeners;
```

Comment:
Why we need to use `return 0 < count($this->listenerIds) || 0 < count($this->listeners)` ?
Do you want be sure, that you will get bool?
But, if I use logical operators (||, &&, and so on) I will get bool.
For example,
`return 10 || 0`; you will not get 10, you will get result of logical operation, it will be `true`, we do not need to convert it to bool.